From 897c5b0963ecbca33459e62549f1b43c75a08880 Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Thu, 2 Oct 2025 15:06:11 +0200
Subject: [PATCH 381/553] QWizard: make QWizardHeader a bit more safe

The Ruler ctor is only used from QWizardRuler, a subclass, so mark it
as protected, because it doesn't initialize any of the data members.

Found while fixing a Coverity complaint in the vicinity.

Amends the start of the public history.

Pick-to: 6.8 6.5
Change-Id: I407940dc8f720976dec0608eae1649d088f6f98d
Reviewed-by: Volker Hilsheimer <volker.hilsheimer@qt.io>
(cherry picked from commit 38a9db9f63a4ebdff86cdf726ea50e55413bcedf)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/widgets/dialogs/qwizard.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/widgets/dialogs/qwizard.cpp b/src/widgets/dialogs/qwizard.cpp
index 44f5185672d..7473fcd2963 100644
--- a/src/widgets/dialogs/qwizard.cpp
+++ b/src/widgets/dialogs/qwizard.cpp
@@ -246,11 +246,12 @@ bool QWizardLayoutInfo::operator==(const QWizardLayoutInfo &other) const
 
 class QWizardHeader : public QWidget
 {
-public:
+protected:
     enum RulerType { Ruler };
 
     inline QWizardHeader(RulerType /* ruler */, QWidget *parent = nullptr)
         : QWidget(parent) { setFixedHeight(2); }
+public:
     QWizardHeader(QWidget *parent = nullptr);
 
     void setup(const QWizardLayoutInfo &info, const QString &title,
-- 
2.51.2

